src/renderer/components/+cluster/cluster-overview.tsx

```diff
@@ -39,14 +41,19 @@ export class ClusterOverview extends React.Component {
 
   render() {
     const isLoaded = nodesStore.isLoaded && podsStore.isLoaded;
+    const isMetricHidden = userStore.isMetricHidden(ResourceType.Cluster);
 
     return (
       <TabLayout>
         <div className="ClusterOverview">
           {!isLoaded ? <Spinner center/> : (
             <>
-              <ClusterMetrics/>
-              <ClusterPieCharts/>
+              {!isMetricHidden &&
+                <ClusterMetrics/>
+              }
+              {!isMetricHidden &&
+                <ClusterPieCharts/>
+              }
               <ClusterIssues/>
             </>
           )}
```

Comment:
This is getting rather complex for inline conditional rendering, would you mind refactoring it out into a seperate function, say `renderClusterOverview()`?

```ts
renderMetrics(isMetricsHidden: boolean) {
  if (isMetricsHidden) {
    return null;
  }

  return (
    <>
      <ClusterMetrics/>
      <ClusterPieCharts/>
    </>
  );
}

renderClusterOverview(isLoaded: boolean, isMetricHidden: boolean) {
  if (!isLoaded) {
    return <Spinner center/>;
  }

  return (
    <>
      {this.renderMetrics(isMetricsHidden)}
      <ClusterIssues/>
    </>
  );
}
```

Reply:
thank you, I'll do